### DISCUSSION TITLE
Help wanted: osu javascript port

### DISCUSSION THREAD

**Discussion by ethanaobrien** [ppy/osu#15254] [Q&A] [answered]
I am working on writing a javascript based osu (wosu) found [here](https://github.com/ethanaobrien/wosu) and my current plan is:
1: Get circles, spinners, and sliders
2. Organize project into separate files (don't store everything in `index.html`) (This should probably be the most important)
3. Clean up UI, points, performance, etc.
4. Go through and bug test everything

I've been running into problems with trying to implement the sliders.
Could someone help me out with trying to port the slider implementation such as helping write the code or by giving sources of where in the original code this is implemented.
I think most of the problem is still that I don't have the math skills yet to know how to calculate the total time and position of the object. I'm just now taking pre-cal (I'm a Junior) so my skills are limited.

It would be greatly appreciated if someone could help me write the code and implement sliders.

Dev branch (shows what I have done in sliders so far): https://github.com/ethanaobrien/wosu/tree/dev
Specific line with render function: https://github.com/ethanaobrien/wosu/blob/dev/index.html#L1079

Why do I want to do this? 
This would make osu playable on every version of linux, windows, macos, chromeos, or anything that has an updated browser.
No having to compile code to run - people just starting out can easily edit (once I separate everything into files)
To learn new things

**Comment by gagahpangeran** [comment, marked as answer]
Sorry, this is not such place for this kind of things. You can try post at https://osu.ppy.sh/community/forums/2

Regarding hit objects (circle, slider, spinner, etc) implementation, all of the source code are in this repository.
